README


This archive contains the distribution Dist-Zilla-Plugin-Test-CheckBreaks,
version 0.011:

  Generate a test that shows what modules you are breaking

This software is copyright (c) 2014 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for Dist-Zilla-Plugin-Test-CheckBreaks

0.011     2014-08-30 03:04:56Z
          - removed bunk test prereqs from version 0.010

0.010     2014-08-27 03:06:27Z
          - inject more accurate test prerequisites into the distribution
          - improved tests and metadata

0.009     2014-07-02 03:25:05Z
          - define a specific version needed for Module::Metadata, for the
            interfaces we use

0.008     2014-06-27 16:35:50Z
          - minor documentation fixes and metadata tweaks

0.007     2014-02-23 00:39:22Z
          - avoid adding an empty x_breaks field to metadata

0.006     2014-02-16 03:02:30Z
          - compatibility fix for older Dist::Zilla (that lacks knowledge of
            file encodings)

0.005     2014-02-12 05:12:42Z
          - fix failing test with old Test::More
          - fix invalid munging of bare version numbers in metadata

0.004     2014-02-11 06:28:29Z
          - keep usage of ::Conflicts module inside an eval to protect against
            blowing up when a conflicting module fails to compile

0.003     2014-02-11 06:13:34Z
          - remove invalid assumption that [Breaks] is installed by inlining a
            stripped down version into the necessary test

0.002     2014-02-10 02:14:48Z
          - fix error in how conflicts module was being loaded in the test

0.001     2014-02-10 01:21:15Z
          - Initial release.


t/lib/Breaks.pm
use strict;
use warnings;
package # hide from PAUSE! but see Dist::Zilla::Plugin::Breaks...
    Breaks;
use Moose;
with 'Dist::Zilla::Role::MetaProvider';
has breaks => (
    is => 'ro', isa => 'HashRef[Str]',
    required => 1,
);
around BUILDARGS => sub {
    my $orig = shift;
    my $self = shift;
    my $args = $self->$orig(@_);
    return {
        zilla => delete $args->{zilla},
        plugin_name => delete $args->{plugin_name},
        breaks => $args,
    };
};
sub metadata
{
    my $self = shift;
    my $reqs = CPAN::Meta::Requirements->new;
    my $breaks_data = $self->breaks;
    $reqs->add_string_requirement($_, $breaks_data->{$_}) foreach keys %$breaks_data;
    return { x_breaks => $reqs->as_string_hash };
}
1;


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-Test-CheckBreaks was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.013
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.24.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.037',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Generate a test that shows what modules you are breaking",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.037"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Test-CheckBreaks",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::Test::CheckBreaks",
  "PREREQ_PM" => {
    "Data::Dumper" => 0,
    "Data::Section" => "0.004",
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "List::Util" => "1.33",
    "Module::Metadata" => "1.000005",
    "Module::Runtime" => 0,
    "Moose" => 0,
    "Path::Tiny" => 0,
    "Sub::Exporter::ForMethods" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "CPAN::Meta::Check" => "0.007",
    "CPAN::Meta::Requirements" => 0,
    "Dist::Zilla::Role::MetaProvider" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::More" => "0.88",
    "if" => 0,
    "lib" => 0
  },
  "VERSION" => "0.011",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::Meta::Check" => "0.007",
  "CPAN::Meta::Requirements" => 0,
  "Data::Dumper" => 0,
  "Data::Section" => "0.004",
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::MetaProvider" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "List::Util" => "1.33",
  "Module::Metadata" => "1.000005",
  "Module::Runtime" => 0,
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Sub::Exporter::ForMethods" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::More" => "0.88",
  "if" => 0,
  "lib" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);





Build.PL
# This Build.PL for Dist-Zilla-Plugin-Test-CheckBreaks was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.005
if (eval 'use Module::Build::Tiny 0.037; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-Test-CheckBreaks was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.006.
    use 5.006;
    # use Module::Build::Tiny 0.037;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.037"
      },
      "dist_abstract" => "Generate a test that shows what modules you are breaking",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-Test-CheckBreaks",
      "dist_version" => "0.011",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::Test::CheckBreaks",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Data::Dumper" => 0,
        "Data::Section" => "0.004",
        "Dist::Zilla::File::InMemory" => 0,
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::FileGatherer" => 0,
        "Dist::Zilla::Role::FileMunger" => 0,
        "Dist::Zilla::Role::PrereqSource" => 0,
        "Dist::Zilla::Role::TextTemplate" => 0,
        "List::Util" => "1.33",
        "Module::Metadata" => "1.000005",
        "Module::Runtime" => 0,
        "Moose" => 0,
        "Path::Tiny" => 0,
        "Sub::Exporter::ForMethods" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "CPAN::Meta::Check" => "0.007",
        "CPAN::Meta::Requirements" => 0,
        "Dist::Zilla::Role::MetaProvider" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "File::pushd" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::More" => "0.88",
        "if" => 0,
        "lib" => 0
      }
    );


    my %fallback_build_requires = (
      "CPAN::Meta::Check" => "0.007",
      "CPAN::Meta::Requirements" => 0,
      "Dist::Zilla::Role::MetaProvider" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "File::pushd" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::More" => "0.88",
      "if" => 0,
      "lib" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
